backend/testfiles/execution/cloud/db.dark

```diff
@@ -750,26 +753,70 @@ module FindAll =
                                                                                       "Ross" ]
 
   // inlining user functions
-  let userAnd (user: Person) (human: Bool) : Bool =
-    PACKAGE.Darklang.Stdlib.Bool.and_v0 user human
+  let userAnd (a: Bool) (b: Bool) : Bool = PACKAGE.Darklang.Stdlib.Bool.and_v0 a b
 
   let userStringContains (user: Person) (str: String) : Bool =
     PACKAGE.Darklang.Stdlib.String.contains_v0 user.name str
 
   let userLessThan (user: Person) (height: Int) : Bool =
     Builtin.Int.lessThan_v0 user height
 
-  let nestedUserFn (user: Person) (height: Int) : Bool =
-    userAnd user.human (userLessThan user.height height)
+  let userAdd (a: Int) (b: Int) : Int = a + b
+
+  (friends (fun p -> userAnd (true || false) (p.height > 66))) = [ " Chandler "
+                                                                   "Ross" ]
+
+  (friends (fun p -> let a = true in (userAnd (a && a) (p.height > 66)))) = [ " Chandler "
+                                                                              "Ross" ]
 
   (friends (fun p -> userAnd p.human true)) = [ " Chandler "; "Rachel"; "Ross" ]
 
   (friends (fun p -> userStringContains p "Chandler")) = [ " Chandler " ]
 
   (friends (fun p -> userLessThan p.height 65)) = [ "GrumpyCat" ]
 
+  (friends (fun p ->
+    let a = 1 in
+    let b = 9 in
+    (p.height == (userAdd 1 b) && (p.age == b)))) = [ "GrumpyCat" ]
+
+  (friends (fun p ->
+    let a = 1 in
+    let b = 9 in
+    (p.height == (userAdd (b - 8) b) && (p.age == b)))) = [ "GrumpyCat" ]
+
+  // TODO: this should work, but it doesn't. There is an error when we try to inline the function's parameters/arguments
+  // (friends (fun p -> let a = 1 in let b = 9 in (p.height == userAdd 6 (b - 5)))) = [ "GrumpyCat" ]
```

Comment:
I think the problem here extends beyond just checking if the 'b' parameter of 'useAdd' shadows the value b = 9, as even the following test fails
``` 
  (friends (fun p ->
    let a = 1 in
    let c = 9 in
    (p.height == userAdd 6 (c - 5)))) = [ "GrumpyCat" ]
   ```
Still trying to figure out how to fix it. All I know so far is where the infinite recursion is happening ↓
   https://github.com/darklang/dark/blob/879e84cf8bb94ae0a25ceac1f16fe349bf1f1154/backend/src/LibExecution/RuntimeTypesAst.fs#L341-L348